src/scripts/party/PartyPokemon.ts

```diff
@@ -53,16 +53,16 @@ class PartyPokemon implements Saveable {
         category = 0,
         pokerus = false
     ) {
-        this.proteinsUsed = ko.observable(proteinsUsed);
-        this.effortPoints = ko.observable(effortPoints);
-        this._breeding = ko.observable(breeding);
-        this._shiny = ko.observable(shiny);
-        this._level = ko.observable(1);
-        this._attackBonusPercent = ko.observable(attackBonusPercent);
-        this._attackBonusAmount = ko.observable(attackBonusAmount);
-        this._attack = ko.observable(this.calculateAttack());
-        this._category = ko.observable(category);
-        this._pokerus = ko.observable(pokerus);
+        this.proteinsUsed = ko.observable(proteinsUsed).extend({ numeric: 0 });
+        this.effortPoints = ko.observable(effortPoints).extend({ numeric: 0 });
+        this._breeding = ko.observable(breeding).extend({ boolean: null });
+        this._shiny = ko.observable(shiny).extend({ boolean: null });
+        this._level = ko.observable(1).extend({ numeric: 0 });
+        this._attackBonusPercent = ko.observable(attackBonusPercent).extend({ numeric: 0 });
+        this._attackBonusAmount = ko.observable(attackBonusAmount).extend({ numeric: 0 });
+        this._attack = ko.observable(this.calculateAttack()).extend({ numeric: 0 });
```

Comment:
This property doesn't look like we update or use it anywhere, unless I failed searching somehow.

Reply:
Only place It seems to be used is `Party.ts → calculateOnePokemonAttack` which doesn't seem right? maybe?
_i'm probably just misinterpreting what it's doing_
```ts
        const pAttack = useBaseAttack ? pokemon.baseAttack : (ignoreLevel ? pokemon.calculateAttack(ignoreLevel) : (pokemon.attack * (this.calculateEVAttackBonus(pokemon))));
```
Also `Egg.ts → hatch`
```ts
            // Recalculate current attack
            partyPokemon.attack = partyPokemon.calculateAttack();
```
And lastly `BreedingController.ts → getDisplayValue` (and the statistics modal)
```ts
            case 'attack': return `Attack: ${pokemon.attack.toLocaleString('en-US')}`;
```
It looks like we could also use it for the sorting instead of recalculating the attack each time?
```diff
    [SortOptions.attack]: {
        text: 'Attack',
-        getValue: (p) => p.calculateAttack(),
+        getValue: (p) => p.attack,
    },
```

Reply:
Ah I think I tried searching with a bracket because it is an observable, forgetting we have getters to avoid doing that...